### ISSUE TITLE
Allow base image from a private repository

### ISSUE THREAD

**Issue by vincent-lhote** [drone-plugins/drone-docker#187]
It’s possible to use a private registry with its credentials, but if the base image uses a private registry that is different from the one used for the new image, this plugin doesn’t provide a way to give the credentials.

I’d like to be able to either:
– use the registry information on the project in drone,
– or provide new variables to use in the `.drone.yml`.

**Comment by tboerger** [comment]
Have you tried to provide registry credentials to drone?

**Comment by vincent-lhote** [comment]
Yes there is registry credentials (in settings/registry) and they do not appear to be used by the plugin. From what I understood of the code, there is only a single `docker login` done and its with the credentials given. It’s not possible to build an image where the base image and the result need different credentials.

**Comment by bradrydzewski** [comment]
the only think I can think of is that we optionally allow the user to provide a `docker_config_json` secret variable, which is essentially just the `.docker/config.json` file, which can store authentication credentials for multiple providers. Then we can write this to the `/root/.docker/config.json`

**Comment by vincent-lhote** [comment]
Yes there is registry credentials (in settings/registry) and they do not appear to be used by the plugin. From what I understood of the code, there is only a single `docker login` done and its with the credentials given. It’s not possible to build an image where the base image and the result need different credentials.

**Comment by vincent-lhote** [comment]
It would be better to use the registry information in drone (…/settings/registry) as the information is already there, just not used. I’m not sure it is given to plugins.

**Comment by natlibfi-arlehiko** [comment]
Any plans for this?

**Comment by poksiala** [comment]
Pulling from private registry seems to be working with this https://discourse.drone.io/t/how-to-pull-private-images-with-1-0/3155

**Comment by natlibfi-arlehiko** [comment]
Yeah, it does now.

**Comment by ywanbing** [comment]
Has this problem been solved?

**Comment by ywanbing** [comment]
@vincent-hatakeyama I have also encountered this problem now, is there any good method?

**Comment by vincent-hatakeyama** [comment]
> [@vincent-hatakeyama](https://github.com/vincent-hatakeyama) I have also encountered this problem now, is there any good method?

We’ve stopped using drone for quite a while. We use heptapod and its CI when bitbucket stopped supporting mercurial repositories.
